### ISSUE TITLE
Argocd outofsync with no diff changes

### ISSUE THREAD

**Issue by andrewhibbert** [argoproj/argo-cd#15898]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

If you have fields that are set to default (see below for examples) argocd will say app is OutOfSync and try to sync even though the diff is blank
```
ignoreDifferences: []
fileParameters: []
ignoreMissingValueFiles: false
passCredentials: false
skipCrds: false
valueFiles: []
```

**To Reproduce**

Enter above fields as defaults instead of leaving them out, argocd will want to sync even no changes

**Expected behavior**

Does not sync

**Screenshots**

**Version**

```
argocd-server: v2.8.4+c279299
```
**Note:** This did not go OutOfSync in 2.7.7

**Logs**

Before updating the yaml file:
[argo-cd-before.txt](https://github.com/argoproj/argo-cd/files/12868543/argo-cd-before.txt)

After, this is when it wants to sync:
[argo-cd-after.txt](https://github.com/argoproj/argo-cd/files/12868544/argo-cd-after.txt)

Application shows blank diff:
![BlankDiff](https://github.com/argoproj/argo-cd/assets/77166944/dbaa2b5c-9f1c-4643-9cee-f12eb8608d99)

**Comment by nicl-dev** [comment]
We are experiencing the same on the same version.

**Comment by nkprince007** [comment]
noticing the same issue with argocd v2.10.7

I have also tried experimenting with the resource tracking using all three methods from this guide: https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/#resource-tracking

Despite my best efforts, it is still showing the same for multiple applications.

```
Status: OutOfSync
Health: Missing
```

**Comment by ro0NL** [comment]
same

**Comment by ro0NL** [comment]
We noticed this issue when doing a fresh rollout. But we migrated from `values: |` to `valuesObject` meanwhile, hence this was super confusing.

Reverting back to `values: |` made argo do a proper sync. Then again reverting back to `valuesObject` seems to work!

**Comment by nkprince007** [comment]
This issue is resolved for me after upgrading to v2.11.0
